src/main/java/bio/terra/cda/app/controller/MetaApiController.java

```diff
@@ -30,16 +36,27 @@ public MetaApiController(ApplicationConfiguration applicationConfiguration) {
 
   @Override
   public ResponseEntity<SystemStatus> serviceStatus() {
-    HttpStatus httpStatus = HttpStatus.OK;
-
-    SystemStatusSystems otherSystemStatus =
-        new SystemStatusSystems().ok(true).addMessagesItem("everything is fine");
+    SystemStatus systemStatus = new SystemStatus();
+    try {
+      BigQuery bigQuery = BigQueryOptions.newBuilder().setProjectId("gdc-bq-sample").build().getService();
+      String StatusCheck = bigQuery.getDataset("cda_mvp").getDatasetId().getDataset();
+      if (StatusCheck.equals("cda_mvp")) {
+        SystemStatusSystems otherSystemStatus = new SystemStatusSystems().ok(true).addMessagesItem("everything is fine");
+        systemStatus.ok(true).putSystemsItem("BigQuery", otherSystemStatus);
+        return new ResponseEntity<>(systemStatus, HttpStatus.OK);
+      }
+    } catch (Exception e) {
+      logger.error(e.getMessage());
+      SystemStatusSystems otherSystemStatus = new SystemStatusSystems();
+      otherSystemStatus.setOk(false);
+      otherSystemStatus.addMessagesItem("Error");
+      systemStatus = new SystemStatus().putSystemsItem("BigQuery", otherSystemStatus);
+      logger.error(e.toString());
+    }
+    return new ResponseEntity<>(systemStatus, HttpStatus.BAD_REQUEST);
```

Comment:
I believe the proper HTTP status code for a status failure is 503, `SERVICE_UNAVAILABLE`